File: Leetcode/125.js
var isPalindrome = function (s) {

  if(s=== ' ') return true
    let match = s.match(/[a-zA-Z0-9]/g);

    let math = match.join("").toLowerCase();

    let l = math.split("").reverse().join("");

   return math === l;
};

let s = "ab_a",
    b = "race a! car";

console.log(isPalindrome(s));
console.log(isPalindrome(b));

File: Leetcode/125.py
import re

class Solution:
    def isPalindrome(self, s: str) -> bool:
        # Use regex to remove non-alphanumeric characters
        filtered = re.findall(r'[a-zA-Z0-9]', s)
        # Convert to lowercase
        processed_string = ''.join(filtered).lower()
        # Check if the string reads the same forward and backward
        return processed_string == processed_string[::-1]

# Example usage
solution = Solution()
print(solution.isPalindrome("A man, a plan, a canal: Panama"))  # Output: True
print(solution.isPalindrome("race a car"))                     # Output: False
print(solution.isPalindrome(" "))                              # Output: True
print(solution.isPalindrome("ab_a"))                           # Output: Trues
